pkg/codegen/utilities.go

```diff
@@ -172,3 +172,37 @@ func ExpandShortEnumName(name string) string {
 	}
 	return name
 }
+
+// A simple in memory file system.
+type Fs struct {
+	m map[string][]byte
+}
+
+// Add a new file to the Fs.
+//
+// Panic if the file is a duplicate.
+func (fs Fs) Add(path string, contents []byte) {
+	fs.ensure()
+	_, has := fs.m[path]
+	contract.Assertf(!has, "duplicate file: %s", path)
+	fs.m[path] = contents
+}
+
+func (fs Fs) ensure() {
```

Comment:
If we're keeping it as a struct, I think the receiver will need to be a pointer, since we're modifying the value of `fs.m`:

```suggestion
func (fs *Fs) ensure() {
```

And probably just make all the methods use the pointer for the receiver for consistency.